cmd/admin.go

```diff
@@ -0,0 +1,152 @@
+	flags.StringVar(&id, "id", "", "ID to assign to the leaderboard.")
+	flags.BoolVar(&authoritative, "authoritative", false, "True if clients may not submit scores directly, false otherwise.")
+	flags.StringVar(&sortOrder, "sort", "descending", "Leaderboard sort order, 'ascending' or 'descending'.")
+	flags.StringVar(&resetSchedule, "reset", "", "Optional reset schedule in CRON format.")
+	flags.StringVar(&metadata, "metadata", "{}", "Optional additional metadata as a JSON string.")
+	flags.StringVar(&metadata, "metadata", "{}", "Optional additional metadata as a JSON string.")
+
+	if err := flags.Parse(args); err != nil {
+		logger.Fatal("Could not parse admin flags.")
+	}
+
+	if dsns == "" {
+		logger.Fatal("Database connection details are required.")
+	}
+
+	query := `INSERT INTO leaderboard (id, authoritative, sort_order, count, reset_schedule, metadata)
+	VALUES ($1, $2, $3, 0, $4, $5)`
+	params := []interface{}{}
+
+	// ID.
+	if id == "" {
+		params = append(params, uuid.NewV4().Bytes())
+	} else {
+		params = append(params, []byte(id))
+	}
+
+	// Authoritative.
+	params = append(params, authoritative)
+
+	// Sort order.
+	if sortOrder == "ascending" {
+		params = append(params, 0)
+	} else if sortOrder == "descending" {
+		params = append(params, 1)
+	} else {
+		logger.Fatal("Invalid sort value, must be 'ascending' or 'descending'.")
+	}
+
+	// Count is hardcoded in the INSERT above.
+
+	// Reset schedule.
+	if resetSchedule != "" {
+		_, err := cronexpr.Parse(resetSchedule)
+		if err != nil {
+			logger.Fatal("Reset schedule must be a valid CRON expression.")
+		}
+		params = append(params, resetSchedule)
+	} else {
+		params = append(params, nil)
+	}
+
+	// Metadata.
+	metadataBytes := []byte(metadata)
+	var maybeJSON map[string]interface{}
+	if json.Unmarshal(metadataBytes, &maybeJSON) != nil {
+		logger.Fatal("Metadata must be a valid JSON string.")
+	}
+	params = append(params, metadataBytes)
+
+	rawurl := fmt.Sprintf("postgresql://%s?sslmode=disable", dsns)
+	url, err := url.Parse(rawurl)
+	if err != nil {
+		logger.Fatal("Bad connection URL", zap.Error(err))
+	}
+
+	logger.Info("Database connection", zap.String("dsns", dsns))
+
+	// Default to "nakama" as DB name.
+	dbname := "nakama"
+	if len(url.Path) > 1 {
+		dbname = url.Path[1:]
+	}
+	url.Path = fmt.Sprintf("/%s", dbname)
+	db, err := sql.Open(dialect, url.String())
+	if err != nil {
+		logger.Fatal("Failed to open database", zap.Error(err))
+	}
+	if err = db.Ping(); err != nil {
+		logger.Fatal("Error pinging database", zap.Error(err))
+	}
+
+	res, err := db.Exec(query, params...)
+	if err != nil {
+		logger.Fatal("Error creating leaderboard", zap.Error(err))
+	}
+	if rowsAffected, _ := res.RowsAffected(); rowsAffected != 1 {
+		logger.Fatal("Error creating leaderboard, unexpected insert result")
+	}
+
+	logger.Info("Leaderboard created, base64 encoded ID: ", zap.String("id", base64.StdEncoding.EncodeToString(params[0].([]byte))))
```

Comment:
`"Leaderboard created, base64 encoded ID: "` -> `"Leaderboard created"`. Zap will add the ID field with value to the end of the string.